apps/signal-store-angular-demo-standalone/src/app/todos-shared/services/todos-api.service.ts

```diff
@@ -0,0 +1,52 @@
+import { inject, Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Todo } from '../models/todo';
+import { Observable, tap } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { ToastrService } from 'ngx-toastr';
+import { altKeyPressed$ } from '../../core/utils';
+import { ErrorHandlerService } from '../../core/services/error-handler.service';
+
+const todoApiUrl = 'api/todos/';
+const failingTodoApiUrl = 'api/todos-not-ok';
+let apiUrl = todoApiUrl;
+
+altKeyPressed$.subscribe(updateApiUrl);
```

Comment:
It's not a good practice to subscribe inside a singleton service, since this subscription will live forever. But for the purpose of this 'hack', I guess it's acceptable.

Reply:
That is debatable, I guess :)

IMO in forever living services there is no fundamental issue with forever living subscriptions. 
I never saw it causing performance/memory issues.

If you create an effect in NgRx/MiniRx you also create a forever living subscription.
The effect subscribes on a specific action. There is no unsubscribe mechanism. 

There has to be a unsub mechanism in destroyable components or services.
And in general you subscribe only when you need the Observable to do something. 

But for this 'hack'  the forever subscription is indeed acceptable.